core/vm/logger.go

```diff
@@ -106,6 +106,8 @@ func (s *StructLog) ErrorString() string {
 type Tracer interface {
 	CaptureStart(env *EVM, from common.Address, to common.Address, create bool, input []byte, gas uint64, value *big.Int)
 	CaptureState(env *EVM, pc uint64, op OpCode, gas, cost uint64, scope *ScopeContext, rData []byte, depth int, err error)
+	CaptureEnter(env *EVM, type_ CallFrameType, from common.Address, to common.Address, input []byte, gas uint64, value *big.Int)
+	CaptureExit(env *EVM, output []byte, gasUsed uint64)
```

Comment:
q: what's the difference with `CaptureEnd`?

Reply:
Agree the difference between `End` and `Exit` is a bit confusing. Looking for better names.

That said, `CaptureEnd` is invoked at the end of the transaction while `CaptureExit` is invoked at the end of internal calls.

Reply:
I see makes sense. It might be useful to add that as a comment on top of each function from the interface 👍